### DISCUSSION TITLE
How to write a custom prompt?

### DISCUSSION THREAD

**Discussion by nareshbhatia** [SBoudrias/Inquirer.js#1111] [Q&A] [answered]
1. Are there any docs explaining how to write a custom prompt?
2. I need a prompt that displays a list of workspaces in a monorepo and allows the user to choose one. Does such a prompt already exist?

P.S. npm and yarn define workspaces in package.json as shown in the example below:

```json
"workspaces": [
  "apps/*",
  "packages/*"
]
```

**Comment by SBoudrias** [comment, marked as answer]
Hey, we do not have documentation on how to achieve this unfortunately... But there's a ton of examples custom prompts [listed in the README](https://github.com/SBoudrias/Inquirer.js#plugins), you could read their code to see how it works.

Aside from that, I think the new (beta) interface of Inquirer core is much much easier to use to write custom prompts. Refers to the prompts in our own monorepo to see the interface in use. The new [checkbox prompt](https://github.com/SBoudrias/Inquirer.js/blob/master/packages/checkbox/index.js) might be a good place to start to see a complex one - or [input prompt](https://github.com/SBoudrias/Inquirer.js/blob/master/packages/input/index.js) for a simpler one, etc.

**Reply by nareshbhatia** [reply]
Thanks @SBoudrias, I will have a look.
